libraries/streaming-e2e-tests/react-app/src/App.js

```diff
@@ -1,12 +1,84 @@
-import React from 'react';
+import React, { Fragment, useCallback, useEffect, useMemo, useReducer } from 'react';
 import ReactWebChat from 'botframework-webchat';
+import { DirectLineStreaming } from 'botframework-directlinejs';
 
-function App({directLine}) {
-  // nonce related to botbuilder-js content security issue #2762:
-  // https://github.com/microsoft/botbuilder-js/issues/2762
-  return (
-    <ReactWebChat directLine={directLine} nonce="a1b2c3d" />
-  );
+const actions = {
+    error: 'error',
+    token: 'token',
+};
+
+function App({ bot }) {
+    if (!bot) {
+        throw new Error('bot prop is required');
+    }
+
+    const hostname = `https://${bot}.azurewebsites.net`;
+
+    const [state, dispatch] = useReducer(
+        (state, action) => {
+            switch (action.type) {
+                case actions.error:
+                    return { ...state, error: action.value, token: null };
+
+                case actions.token:
+                    return { ...state, error: null, token: action.value };
+
+                default:
+                    return state;
+            }
+        },
+        { error: null, token: null }
+    );
+
+    // Synchronize token with hostname
+    useEffect(() => {
+        (async () => {
+            try {
+                const resp = await fetch(`${hostname}/api/token/directlinease`, {
+                    method: 'POST',
+                });
+
+                const text = await resp.clone().text();
+
+                try {
+                    const token = await resp.json();
+                    dispatch({ type: actions.token, value: token.token });
```

Comment:
Use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

```suggestion
                    const { token } = await resp.json();
                    dispatch({ type: actions.token, value: token });
```